### PR TITLE
chore: migrate ErrorBoundary component from jsx to tsx

### PR DESCRIPTION
**Pull request by Damans227**
### SUMMARY

PR for migrating `ErrorBoundary` class based react component from JSX to TSX

### ADDITIONAL INFORMATION

- [x] #10004
- [x] Enhancement (new features, refinement)
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### SOURCES REFFERED

- [Convert React Prop Types to interface in TypeScript](https://www.benmvp.com/blog/react-prop-types-with-typescript/)
- [componentDidCatch() - react documentation](https://reactjs.org/docs/react-component.html#componentdidcatch)
- [Type check for `error` and `info` used as state properties in react error boundary](https://gist.github.com/andywer/800f3f25ce3698e8f8b5f1e79fed5c9c)

### CODE REVIEW

**Comment by Damans227** [comment]
@etr2460 Ptal,thx! Also, Happy New Year.

File: superset-frontend/src/components/ErrorMessage/ErrorAlert.tsx [hunk 1]
```diff
@@ -82,7 +82,7 @@ const LeftSideContent = styled.div`
 
 interface ErrorAlertProps {
   body: ReactNode;
-  copyText?: string;
+  copyText?: string | JSX.Element;
```
Comment (etr2460) [inline superset-frontend/src/components/ErrorMessage/ErrorAlert.tsx:85]: is this change intentional?

File: superset-frontend/src/components/ErrorMessage/ErrorMessageWithStackTrace.tsx [hunk 2]
```diff
@@ -30,8 +30,8 @@ type Props = {
   error?: SupersetError;
   link?: string;
   subtitle?: React.ReactNode;
-  copyText?: string;
-  stackTrace?: string;
+  copyText?: string | JSX.Element;
```
Comment (etr2460) [inline superset-frontend/src/components/ErrorMessage/ErrorMessageWithStackTrace.tsx:33]: same question here

File: superset-frontend/src/components/ErrorBoundary/index.tsx [hunk 3]
```diff
@@ -17,27 +17,30 @@
...
-};
+interface ErrorBoundaryProps {
+  children: React.ReactNode;
+  onError?: Function;
+  showMessage?: boolean;
+}
+
+export default class ErrorBoundary extends React.Component<
+  ErrorBoundaryProps,
+  { error: Error | null; info: React.ErrorInfo | null }
```
Comment (etr2460) [inline superset-frontend/src/components/ErrorBoundary/index.tsx:31]: Maybe let's make this into an `ErrorBoundaryState` too?

**Review by etr2460 — COMMENTED** [review]
2 questions and a comment

File: superset-frontend/src/components/ErrorMessage/ErrorAlert.tsx (hunk 1 above)
Comment (Damans227) [reply superset-frontend/src/components/ErrorMessage/ErrorAlert.tsx:85]: Type for `copyText` becomes invalid if `JSX.Element` type is not set. Following error gets thrown in the `ErrorMessageWithStackComponent`:

```
(JSX attribute) copyText?: string | undefined
Type 'Element' is not assignable to type 'string'.ts(2322)
ErrorMessageWithStackTrace.tsx(33, 3): The expected type comes from property 'copyText' which is declared here on type 'IntrinsicAttributes & Props' 
```

File: superset-frontend/src/components/ErrorBoundary/index.tsx (hunk 3 above)
Comment (Damans227) [reply superset-frontend/src/components/ErrorBoundary/index.tsx]: Yessir!

File: superset-frontend/src/components/ErrorMessage/ErrorAlert.tsx (hunk 1 above)
Comment (etr2460) [reply superset-frontend/src/components/ErrorMessage/ErrorAlert.tsx:85]: Gotcha, thanks!

**Review by etr2460 — APPROVED** [review]
